src/main/java/com/solace/samples/java/patterns/DirectRequestorNonBlocking.java

```diff
@@ -0,0 +1,144 @@
+                                    System.out.println(String.format("NACK for Message %s - %s", userContext, pubSubPlusClientException));
+                                }
+                                if (pubSubPlusClientException instanceof PubSubPlusClientException.TimeoutException) {
+                                    // This handles the situation that the requester application did not receive a reply for the published message within the specified timeout.
+                                    // This would be a good location for implementing resiliency or retry mechanisms.
+                                    System.out.printf("Publishing action timed out without any reply. Error : : %s%n", pubSubPlusClientException);
+                                    System.out.println("Publish timed-out for message with payload :" + outboundMessage.getPayloadAsString());
+
+                                } else {
+                                    throw new RuntimeException(pubSubPlusClientException);
+                                }
+                            }
+                        }
+                        , Topic.of(topicString), REQUEST_TIMEOUT_MS);
+                loopCounter++;     // increment by one
+
+            } catch (final RuntimeException runtimeException) {
+                System.out.printf("### Caught while trying to publisher.publish(): %s%n", runtimeException);
+                isShutdown = true;  // or try to handle the specific exception more gracefully
+            } finally {
+                try {
+                    Thread.sleep(1000);  // do Thread.sleep(0) for max speed
+                    // Note: STANDARD Edition Solace PubSub+ broker is limited to 10k msg/s max ingress
+                } catch (InterruptedException e) {
+                    isShutdown = true;
+                }
+            }
+        }
+    }
+
+    private static Properties setupPropertiesForConnection(final String... args) {
+        final Properties properties = new Properties();
+        properties.setProperty(SolaceProperties.TransportLayerProperties.HOST, args[0]);          // host:port
+        properties.setProperty(SolaceProperties.ServiceProperties.VPN_NAME, args[1]);     // message-vpn
+        properties.setProperty(SolaceProperties.AuthenticationProperties.SCHEME_BASIC_USER_NAME, args[2]);      // client-username
+        if (args.length > 3) {
+            properties.setProperty(SolaceProperties.AuthenticationProperties.SCHEME_BASIC_PASSWORD, args[3]);  // client-password
+        }
+        properties.setProperty(SolaceProperties.TransportLayerProperties.RECONNECTION_ATTEMPTS, "20");  // recommended settings
+        properties.setProperty(SolaceProperties.TransportLayerProperties.CONNECTION_RETRIES_PER_HOST, "5");
+        properties.setProperty(SolaceProperties.ServiceProperties.RECEIVER_DIRECT_SUBSCRIPTION_REAPPLY, String.valueOf(true)); // re-subscribe after reconnect
```

Comment:
I believe the default for this is true...at least I hope it is. If it is true then let's remove this

Reply:
Sorry to disappoint you, but apparently the default is FALSE 
The java api refers to the JCSMP property Subscriptions reapply, which is by default set as FALSE.
![image](https://user-images.githubusercontent.com/112864703/201635473-1f93f00c-2387-4956-81ee-3fb1d8f0801e.png)